ui/ui-components/components/Tabs.tsx

```diff
@@ -3,76 +3,123 @@ import Link from "next/link";
 import { useRouter } from "next/router";
 import { plural } from "pluralize";
 import { capitalize } from "../utils/languageHelper";
+import { useMemo } from "react";
 
 interface Props {
   name: string;
+  parentNames?: string[];
   draftType?: string;
   tabs: string[];
   defaultTab?: string;
-  scopeName?: string;
-  hideScope?: boolean;
+  hideScopeBreadcrumb?: boolean;
+}
+
+interface GrouparooTabTopic {
+  type: string;
+  id: string;
+  href: string;
+  name: string;
 }
 
 const GrouparooTabs: React.FC<Props> = ({
   name,
+  parentNames,
   draftType,
   tabs,
   defaultTab,
-  scopeName,
-  hideScope = false,
+  hideScopeBreadcrumb,
 }) => {
   const router = useRouter();
-  if (!router.pathname) return null;
 
-  const parts = router.asPath.split("/");
-  const scope = parts.slice(0, parts.length - 3).join("/");
-  const topic = parts[parts.length - 3];
-  const id = parts[parts.length - 2];
-  const verb = parts[parts.length - 1].split("?")[0];
+  const parts = router.asPath.split("?")[0].split("/");
+  const pathnameParts = router.pathname?.split("?")[0].split("/");
+  const numParts = parts.length;
+  const verb = parts[numParts - 1];
+
+  defaultTab ??= tabs[0];
+
+  const navigateToTab = useMemo<(tabName: string) => void>(() => {
+    const scope = parts.slice(0, parts.length - 3).join("/");
+    const topic = parts[parts.length - 3];
+    const pathnameScope = pathnameParts
+      .slice(0, pathnameParts.length - 3)
+      .join("/");
+    const pathnameId = pathnameParts[pathnameParts.length - 2];
+    const id = parts[parts.length - 2];
+
+    return (tabName: string) => {
+      if (verb === tabName) return;
 
-  const pathnameParts = router.pathname.split("/");
-  const pathnameScope = pathnameParts
-    .slice(0, pathnameParts.length - 3)
-    .join("/");
-  const pathnameId = pathnameParts[pathnameParts.length - 2];
+      router.push(
+        `${pathnameScope}/${topic}/${pathnameId}/${tabName}`,
+        `${scope}/${topic}/${id}/${tabName}`
+      );
+    };
+  }, [parts, pathnameParts]);
```

Comment:
Looking at this makes me thing it should be a set of useMemo's for scope, topic, etc, and then a useCallback for the actual function.

Reply:
I thought about using a callback hook but scope, topic, etc is not really used anywhere else in the component and it shouldn't. So it made sense to me to just have it run through useMemo once.